File: eBarberShop/eBarberShop.Model/Narudzbe.cs
namespace eBarberShop.Model
{
    public class Narudzbe
    {
        public int NarudzbeId { get; set; }
        public string BrojNarudzbe { get; set; }
        public DateTime DatumNarudzbe { get; set; }
        public decimal UkupanIznos { get; set; }
        public bool Status { get; set; }
        public bool? Otkazano { get; set; }
        public int KorisnikId { get; set; }

        public virtual Korisnici Korisnik { get; set; }

        public string? imeKorisnika => Korisnik?.Ime;
        public string? prezimeKorisnika => Korisnik?.Prezime;
        public string? emailKorisnika => Korisnik?.Email;
        public string? adersaKorisnika => Korisnik?.Adresa;
        public string? brojTelefonaKorisnika => Korisnik?.BrojTelefona;
        public string? mjestoBoravkaKorisnika => Korisnik.Grad?.Naziv;

    }
}

File: eBarberShop/UI/ebarbershop_desktop/lib/models/narudzbe/narudzbe.dart
import 'package:json_annotation/json_annotation.dart';

part 'narudzbe.g.dart';

@JsonSerializable()
class Narudzbe {
  int narudzbeId;
  String brojNarudzbe;
  DateTime datumNarudzbe;
  double ukupanIznos;
  bool status;
  bool? otkazano;
  int korisnikId;
  String? imeKorisnika;
  String? prezimeKorisnika;
  String? emailKorisnika;
  String? adersaKorisnika;
  String? brojTelefonaKorisnika;
  String? mjestoBoravkaKorisnika;

  Narudzbe(
      this.narudzbeId,
      this.brojNarudzbe,
      this.datumNarudzbe,
      this.ukupanIznos,
      this.status,
      this.otkazano,
      this.korisnikId,
      this.imeKorisnika,
      this.prezimeKorisnika,
      this.emailKorisnika,
      this.adersaKorisnika,
      this.brojTelefonaKorisnika,
      this.mjestoBoravkaKorisnika);

  factory Narudzbe.fromJson(Map<String, dynamic> json) =>
      _$NarudzbeFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbeToJson(this);
}
